src/main/java/com/microsoft/aad/msal4j/SilentRequest.java

```diff
@@ -28,11 +28,20 @@
         this.assertion = assertion;
         this.requestAuthority = StringHelper.isBlank(parameters.authorityUrl()) ?
                 application.authenticationAuthority :
-                Authority.createAuthority(new URL(parameters.authorityUrl()));
+                Authority.createAuthority(new URL(enforceTrailingSlash(parameters.authorityUrl())));
 
         if (parameters.forceRefresh()) {
             application.getServiceBundle().getServerSideTelemetry().getCurrentRequest().cacheInfo(
                     CacheTelemetry.REFRESH_FORCE_REFRESH.telemetryValue);
         }
     }
+
+    protected static String enforceTrailingSlash(String authority) {
```

Comment:
Can you just use the method defined in `AbstractClientApplicationBase` instead of duplicating this code? It might also make sense to just move the helper method into `Authority`.

Reply:
I originally just duplicated it because it felt weird for one of the token request classes to be using methods from `AbstractClientApplicationBase`, however moving it all to `Authority` makes a lot of sense and I just made that change in the latest commit.